dioptas/controller/MaskController.py

```diff
@@ -370,35 +375,46 @@ def cosmic_btn_click(self):
 
     def save_mask_btn_click(self):
         img_filename, _ = os.path.splitext(os.path.basename(self.model.img_model.filename))
-        filename = save_file_dialog(self.widget, "Save mask data",
-                                    os.path.join(self.model.working_directories['mask'],
-                                                 img_filename + '.mask'),
-                                    filter='Mask (*.mask)')
+        filename, selected_filter = QtWidgets.QFileDialog.getSaveFileName(
```

Comment:
I replaced `save_file_dialog` with a direct call to `QtWidgets.QFileDialog.getSaveFileName` to retrieve the selected filter. And I use this to toggle the vertical flip. This paves the way to support the same file extensions for both options.

From what I tested with PyQt5 and PyQt6, a 2-tuple is always returned and the call to `set_extension` is not needed

https://github.com/Dioptas/Dioptas/blob/772f6db87a7a4a929de7c90ca58c1c73f81ef6c1/dioptas/widgets/UtilityWidgets.py#L199-L205